datasets/flwr_datasets/merge_resplitter.py

```diff
@@ -0,0 +1,91 @@
+
+    Parameters
+    ----------
+    merge_config: Dict[Tuple[str, ...], str]
+        Dictionary with keys - tuples of the current split names to values - the desired
+        split names
+
+    Examples
+    --------
+    Create new `DatasetDict` with a split name "new_train" that is created as a merger
+    of the "train" and "valid" splits. Keep the "test" split.
+
+    >>> # Assuming there is a dataset_dict of type `DatasetDict`
+    >>> # dataset_dict is {"train": train-data, "valid": valid-data, "test": test-data}
+    >>> merge_resplitter = MergeResplitter(
+    >>>     merge_config={
+    >>>         ("train", "valid"): "new_train",
+    >>>         ("test", ): "test"
+    >>>     }
+    >>> )
+    >>> new_dataset_dict = merge_resplitter(dataset_dict)
+    >>> # new_dataset_dict is
+    >>> # {"new_train": concatenation of train-data and valid-data, "test": test-data}
+    """
+
+    def __init__(
+        self,
+        merge_config: Dict[Tuple[str, ...], str],
+    ) -> None:
+        self._merge_config: Dict[Tuple[str, ...], str] = merge_config
+        self._check_duplicate_desired_splits()
+
+    def __call__(self, dataset: DatasetDict) -> DatasetDict:
+        """Resplit the dataset according to the `merge_config`."""
+        self._check_correct_keys_in_merge_config(dataset)
+        return self.resplit(dataset)
+
+    def resplit(self, dataset: DatasetDict) -> DatasetDict:
+        """Resplit the dataset according to the `merge_config`."""
+        resplit_dataset = {}
+        for divided_from__list, divide_to in self._merge_config.items():
+            datasets_from_list: List[Dataset] = []
+            for divide_from in divided_from__list:
+                datasets_from_list.append(dataset[divide_from])
+            if len(datasets_from_list) > 1:
+                resplit_dataset[divide_to] = datasets.concatenate_datasets(
+                    datasets_from_list
+                )
+            else:
+                resplit_dataset[divide_to] = datasets_from_list[0]
+        return datasets.DatasetDict(resplit_dataset)
+
+    def _check_correct_keys_in_merge_config(self, dataset: DatasetDict) -> None:
+        """Check if the keys in merge_config are existing dataset splits."""
+        dataset_keys = dataset.keys()
+        specified_dataset_keys = self._merge_config.keys()
+        for key_list in specified_dataset_keys:
+            for key in key_list:
+                if key not in dataset_keys:
+                    raise ValueError(
+                        f"The given dataset key '{key}' is not present in the given "
+                        f"dataset object. Make sure to use only the keywords that are "
+                        f"available in your dataset."
+                    )
+
+    def _check_duplicate_desired_splits(self) -> None:
+        """Check for duplicate desired split names."""
+        desired_splits = list(self._merge_config.values())
+        duplicates = [
```

Comment:
Should we also add checks (or throw a warning) if the original splits are present in more than one of the new ones ? For instance the below will run without throwing error:
```
splitter = MergeResplitter(merge_config={('train','test'): 'fused', ('test',): 'test'})
```

Reply:
Good catch. I'm not sure. 
Flexibility: I'd go for a Warning. I think that the most cases it would be a problem, but maybe there are also intended usages of that? I am not sure. 
Usage-wise: That's probably the utility for beginners, so this will likely be a bug -> Error

Reply:
I like the idea of a warning, but still allowing the user to do it.

Reply:
Yeah, maybe warning is the reasonable option here. I agree with Daniel's view of not restricting users with this.